### DISCUSSION TITLE
Hostname Option12 incorrect when hostname uses numbers only

### DISCUSSION THREAD

**Discussion by chrixxxi** [rsmarples/dhcpcd#153] [General]
Hi, i am encountering an Issue with dhcpcd version 9.4.1 where i do start dhcpcd with the following command:
exec dhcpcd -B -K -4 -h "131313" eth0

The hostname in the Hostname Option 12, the value is '^T^T^T'.
I did try compiling dhcpcd with the hostname '141414' hardcoded to see if it could work and it does.

This seems to me like some kind of parsing error when creating the ifo struct.
Can anyone help me with this issue?

Thanks so much in advance.
Br,
Christopher

Edit:
Did fix it by manipulating the hostname argument to: '-h "\"131313\""'

**Comment by ido** [comment]
I'm unable to reproduce on my end.

The parsing for hostname is handled in `if-options.c:L732`. `parse_str` does handle double quotes (`L264`). Could you run dhcpcd in a debugger and breakpoint at `if-options.c:737` (if you're on the `master` branch, that's where `parse_str` is called)? If you can reproduce this bug,  let me know what value `ifo->hostname` shows after `parse_str`? (i.e., is it getting `^T^T^T` or `131313` or something else?)

Please reopen if you can provide more details.
